mmv1/third_party/terraform/services/bigquery/resource_bigquery_job_test.go

```diff
@@ -81,3 +82,88 @@ resource "google_bigquery_job" "job" {
 }
 `, context)
 }
+
+func TestAccBigQueryJob_validationErrors(t *testing.T) {
+	t.Parallel()
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+		"project":       envvar.GetTestProjectFromEnv(),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBigQueryJob_missingProjectId(context),
+				ExpectError: regexp.MustCompile(
+					`Invalid BigQuery job configuration\. You must either:.*Missing or empty projectId`,
+				),
+			},
+			{
+				Config: testAccBigQueryJob_missingDatasetId(context),
+				ExpectError: regexp.MustCompile(
+					`Invalid BigQuery job configuration\. You must either:.*Missing or empty datasetId`,
+				),
+			},
+		},
+	})
+}
+
+func testAccBigQueryJob_missingProjectId(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_bigquery_table" "foo" {
```

Comment:
These are not referenced by the job so can be removed, same with the ones in the other func